index.rst

```diff
@@ -4,9 +4,164 @@ Writeable Butlers for Science Users
+This means that ``QuantumGraph`` generation *probably* already works on ``RemoteButler``, but inefficiently, and to our knowledge this hasn't been tried.
+It would nevertheless be better for ``QuantumGraph`` generation to have an option to avoid temporary tables, by instead re-uploading the data IDs it would select from the temporary table each time they are needed (the ``Query.join_data_coordinates`` method we already have).
+This may actually be faster for ``DirectButler`` as well, since we often want to join in data IDs that are a small subset of those in the temporary table (and a subset that will have already been computed in the client).
+If that turns out to be the case, we should probably drop temporary table support from the query system, as ``QuantumGraph`` generation was the driving user case for it.
+We should probably drop it from ``RemoteButler`` regardless, as emulated support is probably worse than no support.
+
+QuantumBackedButler Transfers from RemoteButler to DirectButler
+---------------------------------------------------------------
+
+After a ``QuantumGraph`` is generated against the official data repository, we need to be able to execute that graph such that:
+
+- input datasets are fetched from the ``RemoteButler`` to the offical data repository (with URL signing as needed);
+- output datasets are written into the datastore root of the personal data repository;
+- the "transfer" stage that inserts rows into the personal data repository's database can (optionally) transfer inputs as well as outputs, including setting up the collection structure in which the inputs were found in the offical data repository.
+
+This requires additional persistent information that would have to be put into the ``QuantumGraph`` or some other file.
+
+Signed URLs in DirectButler
+---------------------------
+
+In order for a personal data repository to reference an official data product (instead of copying it), the ``DirectButler`` client will need to be able to obtain signed URLs from the ``RemoteButler`` server.
+This could be implemented as a new ``Datastore`` subclass (to be chained in with a regular ``FileDatastore`` for the personal datasets in that repository), an extension to the current ``FileDatastore``.
+It could also be implemented directly within ``DirectButler``, but since we want this functionality in ``QuantumBackedButler`` as well, implementing it in ``Datastore`` seems to make more sense.
+
+User Interfaces
+---------------
+
+We need high-level command-line and notebook-friendly tools for at least the following tasks:
+
+- Create a new, empty satellite data repository from the official data repository.
+  This needs to set up a datastore that can reference official datasets (with URL signing), and it may be convenient for it to always register a few instruments and/or skymaps, and maybe even write curated calibrations.
```

Comment:
> maybe even write curated calibrations

Does this mean "automatically preload a basic set of calibrations"?  I was slightly confused by the use of "write" rather than, say, "copy".

Reply:
This is automatically preloading certain calibrations for which the source of truth lives in a `git` (typically `obs_*_data`) package, vs. having to come from some other data repository.